google/cloud/bigtable/tests/data_integration_test.cc

```diff
@@ -598,41 +561,71 @@ TEST_F(DataIntegrationTest, TableReadMultipleCellsBigValue) {
     return;
   }
 
-  std::string const table_id = RandomTableId();
-  auto table = CreateTable(table_id, table_config);
+  // TODO(#2015) - Capture the elapsed time for each operation, so we can figure
+  //   out which one causes the test to timeout.
+  auto start = std::chrono::steady_clock::now();
+  auto elapsed_ms = [&start] {
+    using ms = std::chrono::milliseconds;
+    auto now = std::chrono::steady_clock::now();
+    auto elapsed = now - start;
+    start = now;
+    return std::chrono::duration_cast<ms>(elapsed).count();
+  };
+
+  auto table = GetTable();
+  // TODO(#2015) - Capture elapsed time to troubleshoot flakiness.
+  SCOPED_TRACE("CreateTable() " + std::to_string(elapsed_ms()) + "ms");
 
   std::string const row_key = "row-key-1";
-  // cell vector contains 10 cells of 25 MiB
-  auto const MiB = 1024L * 1024L;
-
-  std::string value((25 * MiB), 'a');
+  // cell vector contains 4 cells of 32 MiB each, or 128 MiB (without
+  // considering any overhead). That is much larger that the default gRPC
+  // message size (~4 MiB), and yet much smaller than the configured message
+  // size (~256MiB). Therefore, the row would not fit in a message if we failed
+  // to change the default configuration, but it is not so large that it will
+  // fail to work if we miss the overhead estimation.
+  auto const MiB = 1024 * 1024UL;
+  auto const cell_size = 32 * MiB;
+  auto const cell_count = 4;
+  // Smaller rows are not a good test, they would pass with the default setting.
+  auto const min_row_size = 10 * 4 * MiB;
```

Comment:
What does `10 * 4` represent here? Or is it just an arbitrary number for testing.

Reply:
Added a better explanation.